common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -16,6 +16,61 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     document
   }
 
+  /////////////////////////////
+  // External video cleaner
+  /////////////////////////////
+
+  "AmpEmbedCleaner" should "replace an iframe in a YouTube video-element with an amp-youtube element" in {
+    val youtubeUrl = "https://www.youtube.com/watch?v=foo_12-34"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$youtubeUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-youtube").size should be(1)
+  }
+
+  "AmpEmbedCleaner" should "not create an amp-youtube element if videoid missing" in {
+    val youtubeUrl = s"http://www.youtube.com/"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$youtubeUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-youtube").size should be(0)
+  }
+
+  "AmpEmbedCleaner" should "replace an iframe in a Vimeo video-element with an amp-vimeo element" in {
+    val vimeoUrl = "https://vimeo.com/1234"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$vimeoUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-vimeo").size should be(1)
+  }
+
+  "AmpEmbedCleaner" should "not create an amp-vimeo element if videoid missing" in {
+    val vimeoUrl = s"http://vimeo.com/"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$vimeoUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-vimeo").size should be(0)
+  }
+
+  "AmpEmbedCleaner" should "be able to create an amp-youtube element and an amp-vimeo in the same document" in {
+    val youtubeUrl = "https://www.youtube.com/watch?v=foo_12-34"
+    val vimeoUrl = "https://vimeo.com/1234"
+    val doc = s"""<html><body><figure class="element-video" data-canonical-url="$youtubeUrl"><iframe></iframe></figure><figure class="element-video" data-canonical-url="$vimeoUrl"><iframe></iframe></figure></body></html>"""
+    val document: Document = Jsoup.parse(doc)
+    val result: Document = clean(document)
+
+    result.getElementsByTag("amp-youtube").size should be(1)
+    result.getElementsByTag("amp-vimeo").size should be(1)
+  }
```

Comment:
everybody loves tests! 💯 

could add some tests for edge cases with the regex if you felt it was required (might not be).  E.g. `youtubercom.com`, `youtube.com.ru`, `fakeyoutube.com` etc.